lib/ret_web/controllers/page_controller.ex

```diff
@@ -139,17 +139,40 @@ defmodule RetWeb.PageController do
   end
 
   def render_index(conn) do
-    index_meta_tags = Phoenix.View.render_to_string(RetWeb.PageView, "index-meta.html", [])
+    app_config_json = Ret.AppConfig.get_config() |> Poison.encode!()
+    app_config_script = "window.APP_CONFIG = JSON.parse('#{app_config_json |> String.replace("'", "\\'")}')"
+
+    index_meta_tags =
+      Phoenix.View.render_to_string(RetWeb.PageView, "index-meta.html",
+        app_config_script: {:safe, "<script>#{app_config_script}</script>"}
+      )
 
     chunks =
       chunks_for_page("index.html", :hubs)
       |> List.insert_at(1, index_meta_tags)
 
+    app_config_csp = "'sha256-#{:crypto.hash(:sha256, app_config_script) |> :base64.encode()}'"
```

Comment:
nice one!

Reply:
also be sure to test in both ff and chrome, i believe there can be weirdness w the expected base64 RFC